packages/block-library/src/navigation/use-navigation-menu.js

```diff
@@ -16,12 +19,13 @@ export default function useNavigationMenu( navigationMenuId ) {
 			const navigationMenuSingleArgs = [
 				'postType',
 				'wp_navigation',
-				navigationMenuId,
+				slug,
 			];
-			const navigationMenu = navigationMenuId
+			const navigationMenu = slug
 				? getEditedEntityRecord( ...navigationMenuSingleArgs )
 				: null;
```

Comment:
Need to create a new entity if slug is set but there is no navigationMenu with that slug